packages/app-accounts/src/index.tsx

```diff
@@ -68,6 +118,129 @@ class AccountsApp extends React.PureComponent<Props, State> {
   selectEdit = (): void => {
     this.setState({ action: 'edit' });
   }
+
+  selectRestore = (): void => {
+    this.setState({ action: 'restore' });
+  }
+
+  createState (current: KeyringPair | null, previous: KeyringPair | null): State {
+    return {
+      action: 'edit',
+      current,
+      editedName: current ? current.getMeta().name || '' : '',
+      isEdited: false,
+      previous
+    };
+  }
+
+  onChangeAccount = (publicKey: Uint8Array): void => {
+    const current = publicKey && publicKey.length === 32
+      ? keyring.getPair(publicKey)
+      : null;
+
+    this.nextState({
+      current
+    } as State);
+
+    this.selectEdit();
+  }
+
+  onCreatorCreate = (publicKey: Uint8Array): void => {
+    const current = publicKey && publicKey.length === 32
+      ? keyring.getPair(publicKey)
+      : null;
+
+    this.nextState({
+      current
+    } as State);
+
+    this.selectEdit();
+  }
+
+  onEditorChangeName = (editedName: string): void => {
+    this.nextState({ editedName } as State);
+  }
+
+  onEditorCommit = (): void => {
+    const { current } = this.state;
+    const { editedName } = this.state;
+
+    if (!current) {
+      return;
+    }
+
+    keyring.saveAccountMeta(current, {
+      name: editedName,
+      whenEdited: Date.now()
+    });
+
+    this.nextState({} as State);
+  }
+
+  onEditorDiscard = (): void => {
+    const { current } = this.state;
+
+    if (!current) {
+      return;
+    }
+
+    this.nextState({
+      editedName: current.getMeta().name
+    } as State);
+  }
+
+  onEditorForget = (): void => {
+    const { current } = this.state;
+
+    if (!current) {
+      return;
+    }
+
+    this.setState(
+      this.createState(null, null),
+      () => {
+        keyring.forgetAccount(
+          current.address()
+        );
+      }
+    );
+  }
+
+  nextState (newState: State = {} as State): void {
```

Comment:
All these new added functionality, revert to what it was so we have self-contained pieces and not a god component.